# Titanic Dataset - Exploratory Data Analysis (EDA)
**Track:** Neurofive ML Track | **Task 1:** Python Environment Setup & First EDA

This notebook performs an initial exploratory analysis on the famous Titanic dataset to understand its shape, data types, missing values, and numerical/categorical distributions.

In [1]:
import pandas as pd
import numpy as np

print(f"Pandas version: {pd.__version__}")
print(f"NumPy version: {np.__version__}")

Pandas version: 2.2.3
NumPy version: 2.1.2


In [2]:
# Load Titanic dataset using pandas
df = pd.read_csv('titanic.csv')
print(f"Successfully loaded dataset with shape: {df.shape}")

Successfully loaded dataset with shape: (891, 12)


## 1. Dataset Preview (`.head()`)

In [3]:
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Thayer)",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


## 2. Dataset Structure & Information (`.info()`)

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


## 3. Statistical Summary (`.describe()`)

In [5]:
df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


## 4. Column Classification & Missing Value Analysis

In [6]:
# Missing values analysis
missing_data = df.isnull().sum()
missing_data = missing_data[missing_data > 0]
print('--- Columns with Missing Values ---')
for col, count in missing_data.items():
    pct = (count / len(df)) * 100
    print(f"{col}: {count} missing ({pct:.2f}%)")

print('\n--- Numerical Columns ---')
print(df.select_dtypes(include=[np.number]).columns.tolist())

print('--- Categorical Columns ---')
print(df.select_dtypes(include=['object', 'category']).columns.tolist())


--- Columns with Missing Values ---
Age: 177 missing (19.87%)
Cabin: 687 missing (77.10%)
Embarked: 2 missing (0.22%)

--- Numerical Columns ---
['PassengerId', 'Survived', 'Pclass', 'Age', 'SibSp', 'Parch', 'Fare']

--- Categorical Columns ---
['Name', 'Sex', 'Ticket', 'Cabin', 'Embarked']


## 5. Data Story: First Impressions & Key Findings

The Titanic dataset contains **891 rows** and **12 columns**, capturing demographic, passenger class, ticket, and survival info. Numerical features include `Age`, `Fare`, `SibSp`, `Parch`, `PassengerId`, `Pclass`, and `Survived`, while `Name`, `Sex`, `Ticket`, `Cabin`, and `Embarked` form the categorical variables. Significant missing values exist in `Cabin` (77.1%) and `Age` (19.87%), with `Embarked` missing just 2 records. Key metrics reveal an overall survival rate of ~38.38% with passenger ages ranging from 0.42 to 80 years old (mean age ~29.7 years). The heavy missingness in `Cabin` suggests it may require dropping or indicator encoding, while `Age` will require median/grouped imputation prior to predictive modeling.